packages/components/psammead-navigation/src/Dropdown/index.jsx

```diff
@@ -0,0 +1,211 @@
+  margin: 0;
+  border: 0;
+
+  &:focus {
+    box-shadow: inset 0 0 0 ${GEL_SPACING_HLF} ${C_WHITE};
+  }
+`;
+
+const DropdownUl = styled.ul`
+  list-style-type: none;
+  margin: 0;
+  padding-right: ${GEL_SPACING};
+  padding-left: ${GEL_SPACING};
+  border-bottom: 0.125rem solid ${C_SHADOW};
+`;
+
+const DropdownLi = styled.li`
+  padding-top: 0.75rem;
+  padding-bottom: 0.75rem;
+  border-bottom: 0.0625rem solid ${C_SHADOW};
+
+  &:last-child {
+    padding-bottom: ${GEL_SPACING_DBL};
+    border: 0;
+  }
+`;
+
+const StyledDropdownLink = styled.a`
+  ${({ script }) => script && getPica(script)};
+  ${({ service }) => service && getSansRegular(service)}
+  color: ${C_WHITE};
+  text-decoration: none;
+
+  &:focus {
+    text-decoration: underline;
+  }
+`;
+
+const StyledCurrentLink = styled.span`
+  ${({ dir }) => getStyles(dir)}
+`;
+
+export const DropdownNavigationLi = ({
+  children,
+  script,
+  currentPageText,
+  active,
+  service,
+  url,
+  dir,
+}) => (
+  <DropdownLi role="listitem">
+    {active && currentPageText ? (
+      <StyledDropdownLink script={script} service={service} href={url}>
+        <StyledCurrentLink
+          dir={dir}
+          // eslint-disable-next-line jsx-a11y/aria-role
+          role="text"
+        >
+          <VisuallyHiddenText>{currentPageText}</VisuallyHiddenText>
+          {children}
+        </StyledCurrentLink>
+      </StyledDropdownLink>
+    ) : (
+      <StyledDropdownLink script={script} service={service} href={url}>
+        {children}
+      </StyledDropdownLink>
+    )}
+  </DropdownLi>
+);
+
+DropdownNavigationLi.propTypes = {
+  children: string.isRequired,
+  url: string.isRequired,
+  script: shape(scriptPropType).isRequired,
+  service: string.isRequired,
+  active: bool,
+  currentPageText: string,
+  dir: oneOf(['ltr', 'rtl']),
+};
+
+DropdownNavigationLi.defaultProps = {
+  active: false,
+  currentPageText: null,
+  dir: 'ltr',
+};
+
+export const CanonicalDropdown = ({ announcedText, onClose, children }) => (
+  <DropdownWrapper>
+    <CrossButton
+      aria-label={announcedText}
+      onClick={onClose}
+      // eslint-disable-next-line react/jsx-boolean-value
+      aria-expanded="true"
+    >
+      {navigationIcons.cross}
+    </CrossButton>
+    <DropdownUl role="list">{children}</DropdownUl>
+  </DropdownWrapper>
+);
+
+CanonicalDropdown.propTypes = {
+  announcedText: string.isRequired,
+  children: node.isRequired,
+  onClose: func.isRequired,
+};
+
+export const AmpDropdown = ({ announcedText, onClose, children }) => (
+  <DropdownWrapper>
+    <CrossButton
+      aria-label={announcedText}
+      on={onClose}
+      // eslint-disable-next-line react/jsx-boolean-value
+      aria-expanded="true"
+    >
+      {navigationIcons.cross}
+    </CrossButton>
+    <DropdownUl role="list">{children}</DropdownUl>
+  </DropdownWrapper>
+);
+
+AmpDropdown.propTypes = {
+  announcedText: string.isRequired,
+  children: node.isRequired,
+  onClose: string.isRequired,
+};
```

Comment:
Since `CannonicalDropdown` and `AmpDropdown` have the same props, we can define a variable to store the propTypes. This will ensure that when a prop is added to `CannonicalDropdown` it also gets added to `AmpDropdown`.

Reply:
Props names are the same, however in `CanonicalDropdown` `onClose` is a function, but in an `AmpDropdown` `onClose` is a string

Reply:
Aaah ok. What do you think about 
```js
const dropdownProps = /* the props */

AmpDropdown.propTypes = {
...dropdownProps
onClose: string
}
```
Am concerned that the `props` on `CannonicalDropdown` and `AmpDropdown` might become out of sync when adding `props` that are common to both.

Reply:
Looks good, will add that now